### DISCUSSION TITLE
Update this for CC: Tweaked?

### DISCUSSION THREAD

**Discussion by Senya29** [m-doescode/cchbm#3] [Features, Suggestions, & Ideas]
Is there any chance you could update this for CC: tweaked?

**Comment by m-doescode** [comment]
It's been a while, but i could certainly give it a try. No promises though.
What features do you need ported? I'm not aware of any HBM version that's compatible with CC Tweaked...

**Reply by Senya29** [reply]
mainly just the launch pad features, i was gonna give a shot at converting it myself but do not know a lot of java.

**Reply by m-doescode** [reply]
Which version of minecraft would this be for?

**Reply by Senya29** [reply]
1.12.2 on forge

**Reply by m-doescode** [reply]
Seems like it works just fine with CC:T, but the mcmod.info required a specific version of CC that was incompatible.

The [latest release](https://github.com/m-doescode/cchbm/releases/tag/v0.2.1) *should* hopefully work. Let me know if I need to make any other changes.

**Reply by Senya29** [reply]
Okay

**Reply by Senya29** [reply]
So i tried it and, can not get it to wrap or find the launch pad, i just get a nil value and the code crashes. If there is something i might be doing wrong let me know.
![image_2024-06-18_045123878](https://github.com/m-doescode/cchbm/assets/105352772/cf000c8c-9988-4aa1-b92b-b7e8ed92d640)
![image_2024-06-18_045050106](https://github.com/m-doescode/cchbm/assets/105352772/47afe898-e476-4db6-b678-6820a704b08c)
![image_2024-06-18_045101797](https://github.com/m-doescode/cchbm/assets/105352772/493a660b-c90a-4fd4-a945-475e4333c54b)

**Reply by m-doescode** [reply]
Ah, the actually tile entity for the launchpad is directly in the middle, so you will have to place it right in front of its center block, intersecting with the model.

Yes, it's a bit awkward, but I couldn't come up with a solution back then.

**Reply by Senya29** [reply]
Okay

**Reply by m-doescode** [reply]
Also, wrap accepts a side, i.e. `peripheral.wrap("back")`, find accepts a type `peripheral.find("launchpad")`

**Reply by Senya29** [reply]
So, i cant place the computer next to the center block is there a specific way to do it or no?

**Reply by m-doescode** [reply]
Why can't you place it next to the center block exactly?

**Reply by Senya29** [reply]
It will not let me place near the model

**Reply by m-doescode** [reply]
Ah I see, looks like that's a compact launcher, I haven't implemented that peripheral yet. I'm busy at the moment, but I can give it a go later today.

**Reply by Senya29** [reply]
okay

**Reply by m-doescode** [reply]
Just made a simple peripheral. Let me know if you need any more functions exposed.

Here are the docs: [compact_launchpad.md](https://github.com/m-doescode/cchbm/blob/86366da73aef5c1415eb629e755d96a3f7557411/docs/peripherals/compact_launchpad.md)

**Reply by Senya29** [reply]
Okay

**Reply by m-doescode** [reply]
Once you've checked out the docs, let me know if I should release it as-is or make any further changes

**Reply by Senya29** [reply]
it seems as the Jar File has not been updated

**Reply by Senya29** [reply]
Oh Okay

**Reply by Senya29** [reply]
Looks good, i dont really see anything that needs added

**Reply by m-doescode** [reply]
[cchbm-0.3.jar.zip](https://github.com/user-attachments/files/15892218/cchbm-0.3.jar.zip)
I wanted to wait until the update had everything you needed before i released it but if you wanna check it out early here is the jar

**Reply by m-doescode** [reply]
Alright! Thanks for letting me know.

**Reply by Senya29** [reply]
appreciate it

**Reply by m-doescode** [reply]
No worries at all! Im happy to help

**Reply by Senya29** [reply]
You should make this a released mod on Curseforge, as I'm sure others would like to see it on there.

**Reply by m-doescode** [reply]
Aw I appreciate that! I'll consider uploading to curseforge, though im personally not too pleased with their management, but itll also surely be available on modrinth too.

**Reply by Senya29** [reply]
I will say i was glad to find this repo, as this would have been a decent project for me, as i would need to learn a lot more Java.

**Reply by m-doescode** [reply]
Well, I hope it stands to be a useful learning resource for the future

**Reply by Senya29** [reply]
![image](https://github.com/m-doescode/cchbm/assets/105352772/368e3ba4-1a84-434e-9eda-d0eed57065f2)

i get this when running the launch command for Compact Launchpad

**Reply by Senya29** [reply]
Full Code: https://pastebin.com/0PvamGxe

**Reply by m-doescode** [reply]
Do you mind sharing your logs?

**Reply by Senya29** [reply]
19.06 19:54:30 [Server] ComputerCraft-Coroutine-2/ERROR [FML]: %minecraftFormattingException caught during firing event net.minecraftforge.event.entity.PlaySoundAtEntityEvent@d522de6:
19.06 19:54:30 [Server] INFO java.lang.NoSuchMethodError: net.minecraft.util.SoundEvent.func_187503_a()Lnet/minecraft/util/ResourceLocation;
19.06 19:54:30 [Server] INFO 	at mca.core.forge.EventHooks.onPlaySoundAtEntityEvent(EventHooks.java:243) ~[EventHooks.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_898_EventHooks_onPlaySoundAtEntityEvent_PlaySoundAtEntityEvent.invoke(.dynamic) ~[?:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) ~[EventBus.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.event.ForgeEventFactory.onPlaySoundAtEntity(ForgeEventFactory.java:498) ~[ForgeEventFactory.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraft.world.World.func_184148_a(World.java:1121) ~[amu.class:?]
19.06 19:54:30 [Server] INFO 	at com.hbm.tileentity.bomb.TileEntityCompactLauncher.launch(TileEntityCompactLauncher.java:193) ~[TileEntityCompactLauncher.class:?]
19.06 19:54:30 [Server] INFO 	at com.hbm.blocks.bomb.CompactLauncher.explode(CompactLauncher.java:159) ~[CompactLauncher.class:?]
19.06 19:54:30 [Server] INFO 	at tech.eglspace.majdstuff.mods.cchbm.peripheral.CompactLaunchPadPeripheral.callMethod(CompactLaunchPadPeripheral.java:51) ~[CompactLaunchPadPeripheral.class:0.3]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.api.multiperipheral.WrappedMultiPeripheral.callMethod(WrappedMultiPeripheral.java:38) ~[WrappedMultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:86) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:75) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.shared.peripheral.modem.wired.WiredModemPeripheral$RemotePeripheralWrapper.callMethod(WiredModemPeripheral.java:328) ~[WiredModemPeripheral$RemotePeripheralWrapper.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.shared.peripheral.modem.wired.WiredModemPeripheral.callMethod(WiredModemPeripheral.java:147) ~[WiredModemPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.api.multiperipheral.WrappedMultiPeripheral.callMethod(WrappedMultiPeripheral.java:38) ~[WrappedMultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:86) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:75) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.apis.PeripheralAPI$PeripheralWrapper.call(PeripheralAPI.java:113) ~[PeripheralAPI$PeripheralWrapper.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.apis.PeripheralAPI.callMethod(PeripheralAPI.java:397) ~[PeripheralAPI.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.lua.CobaltLuaMachine$1.invoke(CobaltLuaMachine.java:239) ~[CobaltLuaMachine$1.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.function.LuaInterpreter.execute(LuaInterpreter.java:491) ~[LuaInterpreter.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.function.LuaInterpretedFunction.resume(LuaInterpretedFunction.java:196) ~[LuaInterpretedFunction.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.debug.DebugFrame.resume(DebugFrame.java:243) ~[DebugFrame.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.LuaThread.loop(LuaThread.java:566) ~[LuaThread.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.LuaThread$1.run(LuaThread.java:463) ~[LuaThread$1.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.lua.CobaltLuaMachine.lambda$null$0(CobaltLuaMachine.java:75) ~[CobaltLuaMachine.class:?]
19.06 19:54:30 [Server] INFO 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275]
19.06 19:54:30 [Server] INFO 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275]
19.06 19:54:30 [Server] INFO 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
19.06 19:54:30 [Server] ComputerCraft-Coroutine-2/ERROR [FML]: %minecraftFormattingIndex: 1 Listeners:
19.06 19:54:30 [Server] ComputerCraft-Coroutine-2/ERROR [FML]: %minecraftFormatting0: NORMAL
19.06 19:54:30 [Server] ComputerCraft-Coroutine-2/ERROR [FML]: %minecraftFormatting1: ASM: mca.core.forge.EventHooks@4ea69cbb onPlaySoundAtEntityEvent(Lnet/minecraftforge/event/entity/PlaySoundAtEntityEvent;)V
19.06 19:54:30 [Server] ComputerCraft-Coroutine-2/ERROR [computercraft]: %minecraftFormattingError calling call on dan200.computercraft.core.apis.PeripheralAPI@2718c954
19.06 19:54:30 [Server] INFO java.lang.NoSuchMethodError: net.minecraft.util.SoundEvent.func_187503_a()Lnet/minecraft/util/ResourceLocation;
19.06 19:54:30 [Server] INFO 	at mca.core.forge.EventHooks.onPlaySoundAtEntityEvent(EventHooks.java:243) ~[EventHooks.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_898_EventHooks_onPlaySoundAtEntityEvent_PlaySoundAtEntityEvent.invoke(.dynamic) ~[?:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) ~[EventBus.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraftforge.event.ForgeEventFactory.onPlaySoundAtEntity(ForgeEventFactory.java:498) ~[ForgeEventFactory.class:?]
19.06 19:54:30 [Server] INFO 	at net.minecraft.world.World.func_184148_a(World.java:1121) ~[amu.class:?]
19.06 19:54:30 [Server] INFO 	at com.hbm.tileentity.bomb.TileEntityCompactLauncher.launch(TileEntityCompactLauncher.java:193) ~[TileEntityCompactLauncher.class:?]
19.06 19:54:30 [Server] INFO 	at com.hbm.blocks.bomb.CompactLauncher.explode(CompactLauncher.java:159) ~[CompactLauncher.class:?]
19.06 19:54:30 [Server] INFO 	at tech.eglspace.majdstuff.mods.cchbm.peripheral.CompactLaunchPadPeripheral.callMethod(CompactLaunchPadPeripheral.java:51) ~[CompactLaunchPadPeripheral.class:0.3]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.api.multiperipheral.WrappedMultiPeripheral.callMethod(WrappedMultiPeripheral.java:38) ~[WrappedMultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:86) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:75) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.shared.peripheral.modem.wired.WiredModemPeripheral$RemotePeripheralWrapper.callMethod(WiredModemPeripheral.java:328) ~[WiredModemPeripheral$RemotePeripheralWrapper.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.shared.peripheral.modem.wired.WiredModemPeripheral.callMethod(WiredModemPeripheral.java:147) ~[WiredModemPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.api.multiperipheral.WrappedMultiPeripheral.callMethod(WrappedMultiPeripheral.java:38) ~[WrappedMultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:86) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at pl.asie.computronics.cc.multiperipheral.MultiPeripheral.callMethod(MultiPeripheral.java:75) ~[MultiPeripheral.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.apis.PeripheralAPI$PeripheralWrapper.call(PeripheralAPI.java:113) ~[PeripheralAPI$PeripheralWrapper.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.apis.PeripheralAPI.callMethod(PeripheralAPI.java:397) ~[PeripheralAPI.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.lua.CobaltLuaMachine$1.invoke(CobaltLuaMachine.java:239) ~[CobaltLuaMachine$1.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.function.LuaInterpreter.execute(LuaInterpreter.java:491) ~[LuaInterpreter.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.function.LuaInterpretedFunction.resume(LuaInterpretedFunction.java:196) ~[LuaInterpretedFunction.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.debug.DebugFrame.resume(DebugFrame.java:243) ~[DebugFrame.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.LuaThread.loop(LuaThread.java:566) ~[LuaThread.class:?]
19.06 19:54:30 [Server] INFO 	at org.squiddev.cobalt.LuaThread$1.run(LuaThread.java:463) ~[LuaThread$1.class:?]
19.06 19:54:30 [Server] INFO 	at dan200.computercraft.core.lua.CobaltLuaMachine.lambda$null$0(CobaltLuaMachine.java:75) ~[CobaltLuaMachine.class:?]
19.06 19:54:30 [Server] INFO 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275]
19.06 19:54:30 [Server] INFO 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275]

**Reply by m-doescode** [reply]
Oddly enough it works perfectly fine for me on hbm 1.8.3a-G. Can you tell me what version of hbm you're using, and what kind of custom nuke you're launching?

(For the record, I tested it on a server, in case that was the issue.)

**Reply by m-doescode** [reply]
Can you share your entire modlist? I believe a mixin may be to blame.

**Reply by Senya29** [reply]
I'm on HBM 1.8.4a-G, and it is a HE Lil Bub

**Reply by Senya29** [reply]
Directory: C:\Users\fireb\curseforge\minecraft\Instances\Modpack\mods


Mode                 LastWriteTime         Length Name                                                                 
----                 -------------         ------ ----                                                                 
d-----         6/13/2024  11:07 PM                1.12.2                                                               
d-----         6/13/2024  11:03 PM                memory_repo                                                          
-a----          6/7/2024  12:17 AM        2958699 ActuallyAdditions-1.12.2-r152.jar                                    
-a----          6/7/2024  12:17 AM          11604 ActuallyBaubles-1.12-1.1.jar                                         
-a----          6/7/2024  12:17 AM         243622 AE2WTLib-1.12.2-1.0.34.jar                                           
-a----          6/7/2024  12:17 AM        4062630 appliedenergistics2-rv6-stable-7.jar                                 
-a----          6/7/2024  12:17 AM         108306 AutoRegLib-1.3-32.jar                                                
-a----          6/7/2024  12:17 AM         108450 Baubles-1.12-1.5.2.jar                                               
-a----          6/7/2024  12:17 AM          77866 BetterBuildersWands-1.12.2-0.13.2.271+5997513.jar                    
-a----          6/7/2024  12:17 AM        5185177 BiblioCraft[v2.4.6][MC1.12.2].jar                                    
-a----          6/7/2024  12:17 AM        4746208 BiomesOPlenty-1.12.2-7.0.1.2445-universal.jar                        
-a----          6/7/2024  12:17 AM        1815047 cc-tweaked-1.12.2-1.89.2.jar                                         
-a----         6/18/2024  11:06 PM          19587 cchbm-0.3.jar                                                        
-a----         6/18/2024   2:48 AM           7446 cfm.cfg                                                              
-a----          6/7/2024  12:17 AM          96859 Chameleon-1.12-4.1.3.jar                                             
-a----          6/7/2024  12:17 AM        4755282 Chisel-MC1.12.2-1.0.2.45.jar                                         
-a----          6/7/2024  12:17 AM         737407 chiselsandbits-14.33.jar                                             
-a----         6/13/2024  11:17 PM          13792 Clumps-3.1.2.jar                                                     
-a----          6/7/2024  12:17 AM         704330 CodeChickenLib-1.12.2-3.2.3.358-universal.jar                        
-a----          6/7/2024  12:17 AM         848187 CoFHCore-1.12.2-4.6.6.1-universal.jar                                
-a----          6/7/2024  12:17 AM         505987 CoFHWorld-1.12.2-1.4.0.1-universal.jar                               
-a----          6/7/2024  12:17 AM        1291533 Computronics-1.12.2-1.6.6.jar                                        
-a----          6/7/2024  12:17 AM         595301 CookingForBlockheads_1.12.2-6.5.0.jar                                
-a----          6/7/2024  12:17 AM         199045 CTM-MC1.12.2-1.0.2.31.jar                                            
-a----          6/7/2024  12:17 AM         647959 EnderCore-1.12.2-0.5.76.jar                                          
-a----          6/7/2024  12:17 AM        8946575 EnderIO-1.12.2-5.3.70.jar                                            
-a----          6/7/2024  12:17 AM         223250 EnderStorage-1.12.2-2.4.6.137-universal.jar                          
-a----          6/7/2024  12:17 AM        1149733 ExtremeReactors-1.12.2-0.4.5.68.jar                                  
-a----          6/7/2024  12:17 AM          26515 FastWorkbench-1.12.2-1.7.3.jar                                       
-a----          6/7/2024  12:17 AM         475994 FluxNetworks-1.12.2-4.1.1.34.jar                                     
-a----          6/7/2024  12:17 AM        5298295 forestry_1.12.2-5.8.2.422.jar                                        
-a----          6/7/2024  12:17 AM        6278399 Forgelin-1.8.4.jar                                                   
-a----          6/7/2024  12:17 AM        1304958 ForgeMultipart-1.12.2-2.6.2.83-universal.jar                         
-a----         6/13/2024  11:15 PM        2439199 furniture-6.3.2-1.12.2.jar                                           
-a----          6/5/2024  11:43 PM       75657429 hbm-1.8.4a-G.jar                                                     
-a----         6/18/2024  11:07 PM        5985435 Immersive Vehicles-1.12.2-22.15.0.jar                                
-a----          6/7/2024  12:17 AM         205922 ImmersiveCables-1.12.2-1.3.2.jar                                     
-a----          6/7/2024  12:17 AM        6360959 ImmersiveEngineering-0.12-98.jar                                     
-a----          6/7/2024  12:17 AM         797724 immersivepetroleum-1.12.2-1.1.10.jar                                 
-a----          6/7/2024  12:17 AM        8698109 industrialcraft-2-2.8.221-ex112.jar                                  
-a----          6/7/2024  12:17 AM        1603134 industrialforegoing-1.12.2-1.12.13-237.jar                           
-a----          6/7/2024  12:17 AM        1229046 IndustrialWires-1.7-35.jar                                           
-a----          6/7/2024  12:17 AM         263258 InventoryTweaks-1.63.jar                                             
-a----          6/7/2024  12:17 AM         465363 IronBackpacks-1.12.2-3.0.8-12.jar                                    
-a----          6/7/2024  12:17 AM         956692 ironchest-1.12.2-7.0.72.847.jar                                      
-a----          6/7/2024  12:17 AM         653210 jei_1.12.2-4.16.1.302.jar                                            
-a----          6/7/2024  12:17 AM        6990968 journeymap-1.12.2-5.7.1.jar                                          
-a----          6/7/2024  12:17 AM          45829 just-enough-harvestcraft-1.12.2-1.7.2.jar                            
-a----          6/7/2024  12:17 AM          22714 JustEnoughEnergistics-1.12.2-1.0.8.jar                               
-a----          6/7/2024  12:17 AM         222270 JustEnoughResources-1.12.2-0.9.2.60.jar                              
-a----          6/7/2024  12:17 AM         843486 Mantle-1.12-1.3.3.55.jar                                             
-a----         6/18/2024  11:06 PM        2459965 MCA-1.12.2-6.1.0-universal.jar                                       
-a----          6/7/2024  12:17 AM         544856 mcjtylib-1.12-3.5.4.jar                                              
-a----          6/7/2024  12:17 AM          51284 moreoverlays-1.15.1-mc1.12.2.jar                                     
-a----          6/7/2024  12:17 AM          75170 MouseTweaks-2.10-mc1.12.2.jar                                        
-a----          6/7/2024  12:17 AM         803400 MrTJPCore-1.12.2-2.1.4.43-universal.jar                              
-a----          6/7/2024  12:17 AM          30009 MTLib-3.0.7.jar                                                      
-a----          6/7/2024  12:17 AM         426226 mysticallib-1.12.2-1.13.0.jar                                        
-a----          6/7/2024  12:17 AM          31782 NotEnoughIDs-1.5.4.4.jar                                             
-a----          6/7/2024  12:17 AM        3217616 OpenBlocks-1.12.2-1.8.1.jar                                          
-a----          6/7/2024  12:17 AM        2639587 OpenModsLib-1.12.2-0.12.2.jar                                        
-a----         6/19/2024   3:38 PM              0 output.txt                                                           
-a----          6/7/2024  12:17 AM         328793 p455w0rdslib-1.12.2-2.3.161.jar                                      
-a----          6/7/2024  12:17 AM        4912890 Pam's HarvestCraft 1.12.2zg.jar                                      
-a----         6/18/2024  11:12 PM        1097627 plethora-1.12.2-1.2.3.jar                                            
-a----          6/7/2024  12:17 AM        2036136 ProjectRed-1.12.2-4.9.4.120-Base.jar                                 
-a----          6/7/2024  12:17 AM          67384 ProjectRed-1.12.2-4.9.4.120-compat.jar                               
-a----          6/7/2024  12:17 AM        1894139 ProjectRed-1.12.2-4.9.4.120-fabrication.jar                          
-a----          6/7/2024  12:17 AM        3923751 ProjectRed-1.12.2-4.9.4.120-integration.jar                          
-a----          6/7/2024  12:17 AM         724477 ProjectRed-1.12.2-4.9.4.120-lighting.jar                             
-a----          6/7/2024  12:17 AM        2596509 ProjectRed-1.12.2-4.9.4.120-mechanical.jar                           
-a----          6/7/2024  12:17 AM         501169 ProjectRed-1.12.2-4.9.4.120-world.jar                                
-a----          6/7/2024  12:17 AM        3877577 Quark-r1.6-179.jar                                                   
-a----          6/7/2024  12:17 AM           7074 QuickLeafDecay-MC1.12.1-1.2.4.jar                                    
-a----          6/7/2024  12:17 AM         293231 randompatches-1.12.2-1.22.1.10.jar                                   
-a----          6/7/2024  12:17 AM          21519 rangedpumps-0.5.jar                                                  
-a----          6/7/2024  12:17 AM          31802 RedstoneFlux-1.12-2.1.1.1-universal.jar                              
-a----          6/7/2024  12:17 AM        2895803 rftools-1.12-7.73.jar                                                
-a----          6/7/2024  12:17 AM         600728 rftoolsctrl-1.12-2.0.2.jar                                           
-a----          6/7/2024  12:17 AM         148083 rftoolspower-1.12-1.2.0.jar                                          
-a----         6/13/2024  11:13 PM         888319 rsgauges-1.12.2-1.2.8.jar                                            
-a----          6/7/2024  12:17 AM         605947 StorageDrawers-1.12.2-5.4.2.jar                                      
-a----          6/7/2024  12:17 AM         402870 StorageDrawersExtras-1.12-3.1.0.jar                                  
-a----          6/7/2024  12:17 AM        4023372 TConstruct-1.12.2-2.13.0.183.jar                                     
-a----          6/7/2024  12:17 AM        1139012 tesla-core-lib-1.12.2-1.0.18.jar                                     
-a----          6/7/2024  12:17 AM         301893 theoneprobe-1.12-1.4.28.jar                                          
-a----          6/7/2024  12:17 AM          98845 ThermalCultivation-1.12.2-0.3.6.1-universal.jar                      
-a----          6/7/2024  12:17 AM         728082 ThermalDynamics-1.12.2-2.5.6.1-universal.jar                         
-a----          6/7/2024  12:17 AM        2345139 ThermalExpansion-1.12.2-5.5.7.1-universal.jar                        
-a----          6/7/2024  12:17 AM        2219736 ThermalFoundation-1.12.2-2.6.7.1-universal.jar                       
-a----          6/7/2024  12:17 AM         182379 ThermalInnovation-1.12.2-0.3.6.1-universal.jar                       
-a----          6/7/2024  12:17 AM          95850 TinkerToolLeveling-1.12.2-1.1.0.jar                                  
-a----          6/7/2024  12:17 AM         156406 torchmaster_1.12.2-1.8.5.0.jar                                       
-a----         6/18/2024  11:08 PM      115685369 UNU Civilian Pack [MTS] 1.12.2-22.13.0-6.3.0.jar                     
-a----         6/18/2024  11:08 PM       30839089 UNU Parts Pack [MTS] 1.12.2-22.13.0-6.5.0.jar                        
-a----          6/7/2024  12:17 AM         733580 valkyrielib-1.12.2-2.0.20.1.jar                                      
-a----          6/7/2024  12:17 AM         221001 WanionLib-1.12.2-2.9.jar                                             
-a----          6/7/2024  12:17 AM         178994 Waystones_1.12.2-4.1.0.jar                                           
-a----          6/7/2024  12:17 AM         242030 WirelessCraftingTerminal-1.12.2-3.12.97.jar                          
-a----          6/7/2024   8:42 PM        1479535 worldedit-forge-mc1.12.2-6.1.10-dist.jar                             
-a----          6/7/2024  12:17 AM         496307 xnet-1.12-1.8.2.jar                                                  
-a----          6/7/2024  12:17 AM         227426 zerocore-1.12.2-0.1.2.9.jar                                          
-a----         6/13/2024  11:16 PM        3875576 [1.12.2] SecurityCraft v1.9.9.jar

**Reply by m-doescode** [reply]
Does the missile launch if you flick a lever next to it normally?

**Reply by m-doescode** [reply]
Additionally, can you share a pastebin link to your entire logs?

**Reply by Senya29** [reply]
It does launch if i use a lever

**Reply by Senya29** [reply]
This was the only part on the logs that was abnormal
https://pastebin.com/ZTjSJVqb

**Reply by Senya29** [reply]
Using a lever had no issues, i will try downgrading to 1.8.3 for HBM

**Reply by m-doescode** [reply]
I realize that, I'd still like the full logs as they contain more useful information

**Reply by Senya29** [reply]
https://paste.shockbyte.com/emoxotilojatepijesux

**Reply by m-doescode** [reply]
I havent been able to test 1.8.4a-G because it refuses to launch and I have no idea why, so that is possible. On the other hand, glancing at the code, there doesn't seem to be anything that should be causing any problems...

**Reply by Senya29** [reply]
Im gonna downgrade it and see if that helps

**Reply by m-doescode** [reply]
Hope it does

**Reply by Senya29** [reply]
It is even doing it on 1.8.3

**Reply by Senya29** [reply]
https://paste.shockbyte.com/fodosedebimogucuzewu

**Reply by m-doescode** [reply]
That's very odd..... Honestly, I wish I know what the problem was but Im completely unable to reproduce the bug myself. If you're able, it'd be helpful if you could send over a zip of the .minecraft directory (remember to delete account json files)

I apologize that I'm unable to help any further.

**Reply by Senya29** [reply]
[okay](https://www.dropbox.com/scl/fi/smc4c1m3au4i5sz81dano/Modpack.zip?rlkey=tlk13a2nmftdygzr22iiftusu&st=cfm3i4zk&dl=0)

**Reply by Senya29** [reply]
Firing with a Lever works just fine its only when calling .launch which is odd, sinch all the other functions work like .getDesignatorPos, and etc. For now ill just use a redstone signal so activate it.

**Reply by m-doescode** [reply]
I tried loading it up but strangely enough, it just.... worked.

**Reply by Senya29** [reply]
i wonder if its a issue with a server file

**Reply by m-doescode** [reply]
Actually, i only tested it in singleplayer and forgot to test it on a local server... that could be a reason.

One solution I could implement would be to just provide a monkey-patch that try-catches the error and falls back to sending a redstone signal to the missile launchpad... its not the prettiest thing, but it might just have to do.

But for now, you can use the redstone api as you mentioned and that should work.

Really sorry that this has been breaking and thank you for your patience and cooperation throughout

**Reply by Senya29** [reply]
Its alright im not worried about it i updated my code locally too catch it and just use the RFTools Redstone transmitters and receivers to send the signal
